README.md

```diff
@@ -69,46 +69,146 @@ fields that can be selected, so implementations must provide this out-of-band. I
+empty array, then the recommended behavior is to include only fields
+`type`, `stac_version`, `id`, `geometry`, `bbox`, `links`, `assets`, and `properties.datetime`; if `properties.datetime` is null, then it is recommended to include `properties.start_datetime` and `properties.end_datetime`.
+These are the default fields to ensure a valid STAC Item is returned by default.
+Implementations may choose to include other properties, e.g., `properties.created`, but the number
+of default properties fields should be kept to a minimum.
+2. If only `include` is specified, these fields should be the only fields included.
+Any additional fields provided beyond those in the `include` list should be kept
+to a minimum, as the caller has explicitly stated they do not need them.
+3. If only `exclude` is specified, the specified fields fields should not be
+included in the response Item entity, but every other field available for the
+Item should be included.
+4. If `exclude` is specified and `include` is null or an empty
+array, then the `exclude` fields should be excluded from the default set.
+5. For nested fields (e.g., `properties.datetime`), the most specific path
+should be honored first, and `include` should be preferred over `exclude`. For
+example:
+    1. If a field is in `exclude`, and a nested field of that field is in
+    `include`, the nested field should be included, but no other nested
+    fields in the field should be included.  For example, if `properties` is
+    excluded and `properties.datetime` is included, then `datetime`
+    should be the only nested field in `properties`.
+    2. If a field is in `include`, and a nested field of that field is in `exclude`, the field
+    should be included, and the nested field should be excluded.  For example,
+    if `properties` is included and `properties.datetime` is excluded, then
+    `datetime` should not be in `properties`.
+6. If the same field is present in both `include` and `exclude`, it should be included.
+
+### `null` vs. empty vs. missing
+
+There is a semantic difference between a missing value (i.e., if `include` is not
+in the JSON object), and a `null` or empty value.  The recommended behavior
+around missing vs. null and empty values is described in [the section
+above](#includeexclude-semantics), but is summarized here for reference. "ALL"
+means that all of the Item's fields should be returned; "DEFAULT" means the
+default set (i.e. the required fields for a valid STAC Item) should be returned.
+
+| include | exclude | returned |
+| -- | -- | -- |
+| missing, `null`, or empty | missing, `null`, or empty | DEFAULT |
+| ["a", "b"] | missing, `null`, or empty | ["a", "b"] |
+| missing | ["a", "b"] | ALL except for ["a", "b"] |
+| `null` or empty | ["a", "b"] | DEFAULT except for ["a", "b"] |
+| ["a", "b"] | ["a"] | ["a", "b"] |
+| ["a.b"] | ["a"] | `a.b` should be the only field in `a` |
+| ["a"] | ["a.b"] | `a` should be included, but it should not include `a.b` |
+
```

Comment:
should it be an error if you include `a` and exclude `a` and `b`? I don't think it should be, but we should be explicit here as to whether it's a 400 or the extra exclude is ignored. I think ignore is the right behavior, since I could imagine querying different datasets, and always including an ignore for a large field that is only on some of them (though explicit includes is always the right way to go).

Reply:
Yeah, we call out [here](https://github.com/stac-api-extensions/fields/pull/5/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R101) that it should be included, and the third-to-last line of the table shows that case.

Reply:
I don't see where that's called out, and [-3] is include a b exclude a

Reply:
Ok, I was misunderstanding, thanks for sticking with me. Yes, IMO it should be silently ignored -- I don't think we define any error-full (i.e. return 400) behavior in this extension, and this seems like too much of an edge case to start doing that. Added text and an example.